### DISCUSSION TITLE
Examples in the Readme

### DISCUSSION THREAD

**Discussion by NoahTheDuke** [JohnnyMorganz/StyLua#497] [General]
Hey there! I'm just getting into Lua and this is very cool. I don't have a codebase I can run this on so would it be possible to add some examples in the readme? That way, folks such as myself can get a quick taste of "how it looks" and decide quickly if they want to use Stylua or not.

**Comment by JohnnyMorganz** [comment]
Sure, I can try to find some examples. It might clog the already large readme, but I'll take a look.

If it helps at all, the snapshot tests highlight the formatting after running stylua - https://github.com/JohnnyMorganz/StyLua/tree/master/tests/snapshots - although, slightly more difficult to glance through.

These 2 snapshots might give the best idea for now:
https://github.com/JohnnyMorganz/StyLua/blob/master/tests/snapshots/tests__standard%40large-example.lua.snap
https://github.com/JohnnyMorganz/StyLua/blob/master/tests/snapshots/tests__standard%40large-example-2.lua.snap
(For Luau: https://github.com/JohnnyMorganz/StyLua/blob/master/tests/snapshots/tests__luau%40large_example.lua.snap)

The "before" version of large example 2 helps to highlight the kind of difference that stylua tries to make: https://github.com/JohnnyMorganz/StyLua/blob/master/tests/inputs/large-example-2.lua

**Reply by NoahTheDuke** [reply]
Cool, thanks for the quick response. I don't think you need anything massive, just a short 4-5 line function that's demonstrates the "happy path".

This is only vaguely related but have you considered writing a Stylua Style Guide, like [Black's](https://github.com/psf/black#the-black-code-style)?

**Reply by JohnnyMorganz** [reply]
I definitely do want to document more, and also create a playground like prettier's (the playground might also be a useful tool to test out stylua and see the output yourself) - #193 

In terms of a style guide, the vast majority of the initial implementation is based off https://roblox.github.io/lua-style-guide/
I've actually somewhat forgotten if there are any deviations from that guide (excluding the config options)
